SimG4CMS/HGCalTestBeam/src/AHCalDetId.cc

```diff
@@ -0,0 +1,62 @@
+#include "SimG4CMS/HGCalTestBeam/interface/AHCalDetId.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <ostream>
+
+const AHCalDetId AHCalDetId::Undefined(0,0,0);
+
+AHCalDetId::AHCalDetId() : DetId() {
+}
+
+AHCalDetId::AHCalDetId(uint32_t rawid) : DetId(rawid) {
+}
+
+AHCalDetId::AHCalDetId(int row, int col, int depth) {
+  int icol  = (col > 0) ? col : 10-col;
+  int irow  = (row > 0) ? row : 10-row;
+  id_       = HcalDetId(HcalOther,irow,icol,depth).rawId();
+}
+
+AHCalDetId::AHCalDetId(const DetId& gen) {
+  if (!gen.null()) {
+    HcalSubdetector subdet=(HcalSubdetector(gen.subdetId()));
+    if (gen.det()!=Hcal || subdet!=HcalOther) {
+      throw cms::Exception("Invalid DetId") << "Cannot initialize AHCalDetId from " << std::hex << gen.rawId() << std::dec; 
+    }  
+  }
+  id_=gen.rawId();
+}
+
+int AHCalDetId::irow() const {
+  int value = HcalDetId(id_).ietaAbs();
```

Comment:
Given multiple uses of `HcalDetId(id_)` in this class's member functions, that object could just be created once in the constructor and stored as a member variable.